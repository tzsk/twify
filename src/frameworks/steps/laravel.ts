import chalk from 'chalk';
import fs from 'fs-extra';
import path from 'path';

export async function setupWelcomePage() {
  const filename = 'welcome.blade.php';
  const welcomeFile = path.join(process.cwd(), `resources/views`, filename);
  const welcomePage = await fs.readFile(welcomeFile, 'utf8');

  console.log(`\n- Setting up ${chalk.blue.bold(filename)}...`);

  await fs.writeFile(
    welcomeFile,
    welcomePage.replace(
      /<\/head>/,
      "\t@vite(['resources/css/app.css'])\n\t</head>"
    )
  );
}
