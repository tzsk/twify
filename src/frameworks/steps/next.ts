import glob from 'glob';
import util from 'util';

export async function resolveCssLocation() {
  const [match] = await util.promisify(glob)('./**/globals.css');

  if (!match) {
    return './styles/globals.css';
  }

  return match;
}
