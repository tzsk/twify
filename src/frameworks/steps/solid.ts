import glob from 'glob';
import util from 'util';

export async function resolveCssLocation() {
  const [match] = await util.promisify(glob)('./src/**/root.css', {
    ignore: ['node_modules'],
  });

  if (!match) {
    return './src/index.css';
  }

  return match;
}
