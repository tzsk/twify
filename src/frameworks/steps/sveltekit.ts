import fs from 'fs-extra';
import path from 'path';
import j from 'jscodeshift';
import chalk from 'chalk';

export async function movePostCSS() {
  const source = path.join(process.cwd(), 'postcss.config.js');
  const destination = path.join(process.cwd(), 'postcss.config.cjs');
  if (fs.existsSync(source)) {
    await fs.move(source, destination, { overwrite: true });
  }
}

export async function setupLayoutFile() {
  const ts = fs.existsSync(path.join(process.cwd(), 'tsconfig.json'));
  const location = 'src/routes/+layout.svelte';
  const file = path.join(process.cwd(), location);

  if (fs.existsSync(file)) {
    console.log(`\n${chalk.blue.bold(location)} already exists.`);
    return;
  }

  console.log(`\n- Creating ${chalk.blue.bold(location)}...`);
  await fs.ensureFile(file);

  const opening = ts ? `<script lang="ts">` : `<script>`;
  const code = `${opening}
  import "../app.css";
</script>

<slot />
`;

  await fs.writeFile(file, code);
}
