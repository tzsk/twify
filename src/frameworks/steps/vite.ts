import chalk from 'chalk';
import fs from 'fs-extra';
import path from 'path';
import j from 'jscodeshift';

function _withCssImport(code: string): string {
  const ast = j.withParser('flow')(code);

  const imports = ast.find(j.ImportDeclaration);

  imports
    .at(imports.length - 1)
    .insertAfter(j.importDeclaration([], j.literal('./style.css')));

  return ast.toSource();
}

export async function setupMainFile() {
  const ts = fs.existsSync(path.join(process.cwd(), 'tsconfig.json'));
  let filename: string = ts ? 'main.ts' : 'main.js';
  filename = fs.existsSync(path.join(process.cwd(), 'src', filename))
    ? filename
    : `${filename}x`;

  console.log(`\n- Setting up ${chalk.blue.bold(`src/${filename}`)}...`);
  const filePath = path.join(process.cwd(), 'src', filename);

  const code = await fs.readFile(filePath, 'utf8');

  await fs.writeFile(filePath, _withCssImport(code));
}
