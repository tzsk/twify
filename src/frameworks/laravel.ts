import { Framework } from '../types';
import { setupWelcomePage } from './steps/laravel';

const Laravel: Framework = {
  requiredDependencies: ['tailwindcss', 'postcss', 'autoprefixer'],
  initCommands: ['npx tailwindcss init -p'],
  cssLocation: './resources/css/app.css',
  content: [
    './resources/**/*.blade.php',
    './resources/**/*.{js,jsx,ts,tsx,vue,svelte}',
  ],
  steps: [setupWelcomePage],
};

export default Laravel;
