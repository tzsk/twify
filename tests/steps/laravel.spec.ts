import fs from 'fs-extra';
import { setupWelcomePage } from '../../src/frameworks/steps/laravel';

async function runStep(file: string) {
  const code = fs.readFileSync(`./tests/fixtures/laravel/${file}`, 'utf8');
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  vi.spyOn(fs, 'readFile').mockResolvedValue(code as any);
  const writeSpy = vi.spyOn(fs, 'writeFile').mockResolvedValue();

  await setupWelcomePage();

  const [fileName, content] = writeSpy.mock.lastCall || [];

  return { fileName, content };
}

describe('Vite Steps', () => {
  beforeAll(() => {
    vi.stubGlobal('console', { ...console, log: vi.fn() });
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  afterAll(() => {
    vi.resetAllMocks();
  });

  it('can setup laravel project', async () => {
    const { fileName, content } = await runStep('welcome.stub');

    expect(fileName).toMatch('welcome.blade.php');
    expect(content).toMatchSnapshot();
  });
});
