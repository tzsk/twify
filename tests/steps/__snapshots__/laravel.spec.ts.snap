// Vitest Snapshot v1

exports[`Vite Steps > can setup laravel project 1`] = `
"<html>
  <head>
    <title>Laravel</title>
  	@vite(['resources/css/app.css'])
	</head>
  <body>
  </body>
</html>"
`;
